src/snowflake/cli/plugins/nativeapp/manager.py

```diff
@@ -564,12 +579,75 @@ def deploy(
             self._apply_package_scripts()
 
             # 3. Upload files from deploy root local folder to the above stage
+            stage_fqn = stage_fqn or self.stage_fqn
             diff = self.sync_deploy_root_with_stage(
                 bundle_map=bundle_map,
                 role=self.package_role,
                 prune=prune,
                 recursive=recursive,
+                stage_fqn=stage_fqn,
                 local_paths_to_sync=local_paths_to_sync,
             )
 
+        if validate:
+            self.validate(use_scratch_stage=False)
+
         return diff
+
+    def validate(self, use_scratch_stage: bool = False):
+        """Validates Native App setup script SQL."""
+        cc.step(f"Validating Snowflake Native App setup script.")
+        validation_result = self.get_validation_result(use_scratch_stage)
+
+        # First print warnings, regardless of the outcome of validation
+        for warning in validation_result.get("warnings", []):
+            cc.warning(_validation_item_to_str(warning))
+
+        # Then raise an exception if validation failed
+        if validation_result["status"] == "FAIL":
+            messages = [
+                _validation_item_to_str(error)
+                for error in validation_result.get("errors", [])
+            ]
+            raise SetupScriptFailedValidation(messages)
```

Comment:
there was no specific reason, but looking back I like it like that since it separates errors and warnings, and it makes the errors really obvious by printing them inside the ASCII box art (see the screenshot in the PR description). I'm not opposed to changing it if it's a blocker

Reply:
I think we'll want to control how we format these messages, and this way of outputting them will prevent us from doing that. I'll do a pass on the overall output this week or next week, it's long overdue. We should sync on that, but I'm fine if validate becomes slightly inconsistent (but better) while the other commands catch up.

IMHO we keep the box small and generic here (something like "Validation failed, re-run with --no-validate to skip this check"). We output errors the same way as we do warnings, but maybe we make them visually stand out in the output (maybe we use shell colors to highlight errors?)

Reply:
I updated it to print the messages before the error: 
<img width="1715" alt="image" src="https://github.com/snowflakedb/snowflake-cli/assets/171080004/eb441da2-4565-4cad-8713-554a86447a57">